plugins/by-name/ltex-extra/default.nix

```diff
@@ -43,26 +43,60 @@ lib.nixvim.plugins.mkNeovimPlugin {
     ] "Log level.";
   };
 
-  extraConfig = cfg: {
-    warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" {
-      when = !config.plugins.lsp.enable;
-      message = ''
-        You have enabled `ltex-extra` but not the lsp (`plugins.lsp`).
-        You should set `plugins.lsp.enable = true` to make use of the LTeX_extra plugin's features.
+  extraConfig =
+    cfg:
+    let
+      inherit (config.plugins.lsp.servers) ltex;
+      attachLua = ''
+        require("ltex_extra").setup(${lib.nixvim.toLuaObject cfg.settings})
       '';
-    };
-
-    plugins.lsp = {
-      servers.ltex = {
-        # Enable the ltex language server
-        enable = true;
+    in
+    {
+      warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" [
+        {
+          when = !config.plugins.lsp.enable;
+          message = ''
+            You have enabled `ltex-extra` but not the lsp (`plugins.lsp`).
+            You should set `plugins.lsp.enable = true` to make use of the LTeX_extra plugin's features.
+          '';
+        }
+        {
+          # TODO: Added 2025-03-30; remove after 25.05
+          # Warn if servers.ltex seems to be configured outside of ltex-extra
+          when =
+            !ltex.enable
+            && (
+              ltex.onAttach.function != attachLua
+              || ltex.onAttach.override != false
+              ||
+                [ ] != collect (v: !(isAttrs v || v == null)) (
+                  # Remove options that wouldn't be null in an empty ltex config
+                  removeAttrs ltex [
+                    "extraSettings"
+                    "package"
+                    "onAttach"
+                    "enable"
+                  ]
+                )
```

Comment:
It would be helpful if we told users _which_ options they have configured to trigger this warning, and ideally in which files those options definitions are located.

If that's too much effort for now, could you leave a TODO comment and/or open an issue?